### PR TITLE
refactor: rename `TuiTablePaginationModule` to `TuiTablePaginationComponent`

### PR DESCRIPTION
**Pull request by splincode**

### CODE REVIEW

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/demo/src/modules/components/table-pagination/examples/import/import.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+// ...
+
+@NgModule({
+  imports: [
+    // ...
+    TuiTablePaginationComponent,
+  ],
+})
+export class YourModule {}
```
Comment (vladimirpotekhin) [inline projects/demo/src/modules/components/table-pagination/examples/import/import.md:12]: nit: in other docs we use `MyModule` or `MyComponent`, not `Your` 😄

Comment (splincode) [reply projects/demo/src/modules/components/table-pagination/examples/import/import.md]: fixed

**Comment by nsbarsukov** [comment]
@splincode 
Fix build, please

**Comment by splincode** [comment]
@nsbarsukov fixed

File: projects/addon-table/components/table-pagination/table-pagination.options.ts [hunk 2]
```diff
@@ -7,6 +7,11 @@ function defaultSizeOptionContent({$implicit}: TuiContext<number>): string {
     return `${$implicit}`;
 }
 
+export interface TuiTablePagination {
+    readonly page: number;
+    readonly size: number;
+}
```
Comment (nsbarsukov) [inline projects/addon-table/components/table-pagination/table-pagination.options.ts:13]: This interface is not used anywhere inside `table-pagination.options.ts`. Should we return it back to `table-pagination.component.ts` ?

File: projects/demo/src/modules/components/table-pagination/examples/import/import.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+// ...
+
+@NgModule({
+  imports: [
+    // ...
+    TuiTablePaginationComponent,
+  ],
+})
+export class MyModule {}
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/table-pagination/examples/import/import.md:12]: Let's show modern approach 

```suggestion
@Component({
  imports: [
    // ...
    TuiTablePaginationComponent,
  ],
})
export class MyComponent {}
```

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-table/components/table-pagination/table-pagination.options.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-table/components/table-pagination/table-pagination.options.ts]: fixed

**Pull request merged**